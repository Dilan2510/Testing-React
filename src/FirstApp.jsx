import PropTypes from "prop-types";

// const getNewMessage = (e) => {
//   return e;
// };

// const newMessage = [1, 2, 3, 4, 5, 6, 7, 8, 9];
// const newMessage2 = {
//   message: "Codigo 14",
//   title: "Inbdila",
// };

export const FirstApp = ({ title, subTitle }) => {
  // console.log(props);

  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <h1> {getNewMessage(4)} </h1> */}
      {/* <h1>{JSON.stringify(newMessage2)}</h1> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  subTitle: "Soy un subTitulo",
};
