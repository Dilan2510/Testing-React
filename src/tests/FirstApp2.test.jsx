import { render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe("testing en <FirstApp2>", () => {
  const title = "Hola soy Goku";
  const subtitle = "Soy un subtitulo";

  test("testing for consumir el maatch en el snapshot ", () => {
    const { container } = <FirstApp title={title} />;
    expect(container).toMatchSnapshot();
  });

  test("pasar la etiqueta de h1 ", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(title);
  });

  test("Debe consumir y pasar el titulo ", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe consumir y pasar el titulo ", () => {
    render(<FirstApp subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
