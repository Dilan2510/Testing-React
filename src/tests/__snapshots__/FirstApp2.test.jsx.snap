// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en <FirstApp2> testing for consumir el maatch en el snapshot  1`] = `undefined`;
